src/index.js

```diff
@@ -62,7 +62,9 @@ function _request(path, method, init) {
         ...init
     }).then(response =>
         response.text().then(body => {
-            if (!response.ok) {
+            if (response.status === 401 && retry && credentialResolver) {
+                return credentialResolver(() => _request(path, method, init, credentialResolver, false))
```

Comment:
This looks really bad imho. I'd rather see that we wrap this type of functionality as an "aspect", i.e. wrap the function that manages the credentials with the function that manages the actual request.